### ISSUE TITLE
Instability

### ISSUE THREAD

**Issue by john1761** [kevinlawler/kona#623]
There appears to be an instability. As far as I can tell it requires:
- an inner function which uses the over adverb
- an outer function which calls the inner function at least twice

This seems to be a regression in that the behaviour was not present in interpreters up until early 2019. I am currently using windows only, I do not know if it exists on other platforms.

Example; bug.k:
```
s:{+/!x}         / needs to include the '/ over' adverb, any verb will do
test:{s[x]+s[x]} / needs to call the other function at least twice

_k               / version check - appears to regress in 2019
/s[3]            / uncomment this line and the program does not crash.
"before"
test 3
"after"
```

produces output:
```
C:\bin\k : k.exe bug.k
kona      \ for help. \\ to exit.

"2021-12-24"
"before"

C:\bin\k :
```
The 'after' is not displayed and buried in the windows event log is an unhelpful message about a crash.

Oddly, calling the inner function from global scope before it is called from function scope appears to prevent the crash. Presumably it is accidentally tidying a memory issue?

**Comment by gitonthescene** [comment]
I tested on macOS and I also get the crash with the line commented and not without.
```
➜  kona git:(master) ✗ ./k bug.k 
kona      \ for help. \\ to exit.

"2021-12-06"
"before"
[1]    72342 segmentation fault  ./k bug.k
➜  kona git:(master) ✗ uname -a
Darwin Macintosh.local 21.1.0 Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64 x86_64
➜  kona git:(master) ✗ 
```

Just to make it easier, I've attached the script in the comment above (with the suffix changed): [bug.k.txt](https://github.com/kevinlawler/kona/files/8123545/bug.k.txt)

**Comment by tavmem** [comment]
Got a similar result in Linux
```
$ ./k bug.k 
kona      \ for help. \\ to exit.

"2022-08-30"
"before"
Segmentation fault (core dumped)
```

**Comment by tavmem** [comment]
in k2.8
```
$ ./k bug.k
K 2.8 2000-10-10 Copyright (C) 1993-2000 Kx Systems
Evaluation. Not for commercial use. 
\ for help. \\ to exit.

"2.8"
"before"
6
"after"
```

**Comment by tavmem** [comment]
The problem first appears in the commit of Sep 1, 2019
which was made by TomDuCoin
to fix issue #521: Recursive call overrides earlier data

**Comment by tavmem** [comment]
This area of the code was then modified further (by me) in the commit of Nov 2, 2019
titled "fix #566 and fix #549: recursions housed in subfunctions"
Obviously, the "instability" side effect remained.
